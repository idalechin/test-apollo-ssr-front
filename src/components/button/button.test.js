import React from 'react';
import {shallow} from 'enzyme';
import Button from './index'
import renderer from 'react-test-renderer';
import testIcon from './Rolling.svg'

const btnText = 'Check button'
const className = 'btn-test'

it('Test Button renders right', () => {
  const btn = renderer
    .create(<Button text={btnText} icon={testIcon} className={className}/>)
    .toJSON();
  expect(btn).toMatchSnapshot();
});

it('Test Button text', () => {
  const button = shallow(<Button text={btnText}/>);
  expect(button.text()).toEqual(btnText);
})