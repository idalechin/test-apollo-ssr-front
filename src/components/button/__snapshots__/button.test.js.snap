// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button renders right 1`] = `
<button
  className="btn btn-test  "
  onClick={[Function]}
>
  <svg
    className="btn__icon"
    viewBox="undefined"
  >
    <use
      xlinkHref="#undefined"
    />
  </svg>
  <span
    className="btn__text"
  >
    Check button
  </span>
</button>
`;
